src/defrag.c

```diff
@@ -283,7 +279,7 @@ static void activeDefragZsetNode(void *privdata, void *entry_ref) {
         zskiplistNode *next = x->level[i].forward;
         while (next &&
                (next->score < score ||
-                (next->score == score && sdscmp(next->ele, ele) < 0))) {
+                (next->score == score && sdscmp(zslGetNodeElement(next), ele) < 0))) {
```

Comment:
Not related to your PR, but I think this code could be optimized.
There is no real need to prepare the update array in case the node is not really relocated. we already know the old node pointer so there is no need to do string comparisons. I think we can just compare the next to the old allocation.